### PR TITLE
Adcm 1843 add ui tests for footer and header

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

File: tests/ui_tests/utils.py [hunk 1]
```diff
@@ -88,3 +91,20 @@ class FieldDefinition(BundleObjectDefinition):
...
+
+
+@allure.step('Wait for a new window after action')
+@contextmanager
+def wait_for_new_window(driver: WebDriver, wait_time: int = 10):
+    tabs = driver.window_handles
+    yield
+    WDW(driver, wait_time).until(EC.new_window_is_opened(tabs))
+    tabs = driver.window_handles
+    driver.switch_to.window(tabs[len(tabs) - 1])
```
Comment (dgusakov) [inline tests/ui_tests/utils.py:103]: This method needs a detailed docstring

**Review by skhomuti — APPROVED** [review]

File: tests/conftest.py [hunk 2]
```diff
@@ -208,3 +209,13 @@ def bundle_archive(request, tmp_path):
...
     """
     return _pack_bundle(request.param, tmp_path)
+
+
+@pytest.fixture(scope="function")
+def auth_to_adcm(app_fs, adcm_credentials):
+    """Perform login on Login page ADCM
+    :param app_fs:
+    :param adcm_credentials:
+    """
```
Comment (madlabman) [inline tests/conftest.py:219]: Remove default scope.
Add type hints and remove `:param` directives.

File: tests/ui_tests/app/page/common/header.py [hunk 3]
```diff
@@ -42,3 +43,17 @@ class AuthorizedHeaderLocators(CommonHeaderLocators):
     failed_job_button = Locator(By.XPATH, "//button[@mattooltip='Show failed jobs']", "Header button for failed Jobs")
     help_button = Locator(By.XPATH, "//button[@adcm_test='help']", "Header button for help")
     account_button = Locator(By.XPATH, "//button[@adcm_test='account']", "Header button for account settings")
+    block = Locator(By.XPATH, "//div[contains(@class, 'mat-menu-content')]", "Header popup block")
```
Comment (madlabman) [inline tests/ui_tests/app/page/common/header.py:46]: `block` name looks too broad.

File: tests/ui_tests/app/page/common/header.py [hunk 4]
```diff
@@ -42,3 +43,17 @@ class AuthorizedHeaderLocators(CommonHeaderLocators):
...
+
+        ask_link = Locator(By.XPATH, "//a[@adcm_test='ask_for_help']", "Ask for help link")
+        doc_link = Locator(By.XPATH, "//a[@adcm_test='dock']", "Documentation link")
+
+    class AccountPopup:
+        """ADCM header popup with account settings"""
+
+        settings_link = Locator(By.XPATH, "//a[@adcm_test='settings']", "Settings link")
+        profile_link = Locator(By.XPATH, "//button[@adcm_test='profile']", "Profile link")
+        logout = Locator(By.XPATH, "//button[@adcm_test='logout']", "logout link")
```
Comment (madlabman) [inline tests/ui_tests/app/page/common/header.py:59]: `logout_link` for consistency.

File: tests/ui_tests/test_field_ui_options.py [hunk 5]
```diff
@@ -47,15 +47,15 @@ def test_ui_option_invisible_false_advanced_true(
...
+            'Check that field is not visible by default but with enabled advanced visible'
     ):
         for group in groups:
             assert group.is_displayed(), group.get_attribute("class")
 
 
 @parametrize_by_data_subdirs(__file__, "invisible_false", 'advanced_false')
 def test_ui_option_invisible_false_advanced_false(
-    sdk_client_fs: ADCMClient, path, app_fs, login_to_adcm
+        sdk_client_fs: ADCMClient, path, app_fs, login_to_adcm
```
Comment (madlabman) [inline tests/ui_tests/test_field_ui_options.py:58]: Formatting?

File: tests/ui_tests/test_header_and_footer.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    job_page = JobListPage(app_fs.driver, app_fs.adcm.url)
+    header.wait_url_contains_path(job_page.path)
+    header.check_job_popup()
+
+
+def test_check_header_help_links_for_authorised_user(app_fs, auth_to_adcm):
+    params = {
+        "help_link": "t.me/joinchat/",
+        "docs_link": "docs.arenadata.io/adcm/"
+    }
```
Comment (madlabman) [inline tests/ui_tests/test_header_and_footer.py:73]: Move outside test function.

File: tests/ui_tests/test_header_and_footer.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    header.click_profile_link_in_acc_popup()
+    header.wait_url_contains_path(ProfilePage(app_fs.driver, app_fs.adcm.url).path)
+
+    header.click_account_button_in_header()
+    header.click_logout_in_acc_popup()
+    header.wait_url_contains_path(LoginPage(app_fs.driver, app_fs.adcm.url).path)
+
+
+def test_check_footer_for_authorised_user(app_fs, auth_to_adcm):
+    params = {"docs": "docs.arenadata.io/adcm/notes"}
```
Comment (madlabman) [inline tests/ui_tests/test_header_and_footer.py:106]: Same as above.

File: tests/ui_tests/utils.py [hunk 8]
```diff
@@ -88,3 +91,20 @@ class FieldDefinition(BundleObjectDefinition):
...
+
+
+@allure.step('Close current tab')
+def close_current_tab(driver: WebDriver):
+    tabs = driver.window_handles
+    driver.close()
+    driver.switch_to.window(tabs[0])
```
Comment (madlabman) [inline tests/ui_tests/utils.py:110]: `tabs[0]` exists in any case?

File: tests/ui_tests/test_field_ui_options.py (hunk 5 above)
Comment (annademina) [reply tests/ui_tests/test_field_ui_options.py:58]: pylint

File: tests/ui_tests/test_header_and_footer.py (hunk 6 above)
Comment (annademina) [reply tests/ui_tests/test_header_and_footer.py:73]: why? now this is just for this func

File: tests/ui_tests/test_header_and_footer.py (hunk 7 above)
Comment (annademina) [reply tests/ui_tests/test_header_and_footer.py:106]: this is just for this func

File: tests/ui_tests/utils.py (hunk 8 above)
Comment (annademina) [reply tests/ui_tests/utils.py:110]: otherwise browser isn't opened

File: tests/ui_tests/app/page/common/header.py (hunk 4 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/header.py:59]: it's not a link, it's something like a button

File: tests/ui_tests/app/page/common/header.py (hunk 3 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/header.py:46]: no. it will be like AuthorizedHeaderLocators.block

File: tests/ui_tests/utils.py (hunk 1 above)
Comment (annademina) [reply tests/ui_tests/utils.py]: added

File: tests/conftest.py (hunk 2 above)
Comment (annademina) [reply tests/conftest.py]: removed

File: tests/ui_tests/app/page/common/header.py (hunk 3 above)
Comment (madlabman) [reply tests/ui_tests/app/page/common/header.py:46]: `Header popup ...` confused me.

File: tests/ui_tests/app/page/common/header.py (hunk 4 above)
Comment (madlabman) [reply tests/ui_tests/app/page/common/header.py:59]: So, may be `logout_button`?)

File: tests/ui_tests/test_field_ui_options.py (hunk 5 above)
Comment (madlabman) [reply tests/ui_tests/test_field_ui_options.py:58]: ?

File: tests/ui_tests/test_header_and_footer.py (hunk 6 above)
Comment (madlabman) [reply tests/ui_tests/test_header_and_footer.py:73]: To keep test data separate of test itself.

File: tests/ui_tests/test_field_ui_options.py (hunk 5 above)
Comment (annademina) [reply tests/ui_tests/test_field_ui_options.py:58]: pylint showed this to fix

File: tests/ui_tests/test_header_and_footer.py (hunk 6 above)
Comment (annademina) [reply tests/ui_tests/test_header_and_footer.py:73]: again why?

File: tests/ui_tests/app/page/common/header.py (hunk 4 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/header.py]: ok, changed)

File: tests/ui_tests/test_field_ui_options.py (hunk 5 above)
Comment (madlabman) [reply tests/ui_tests/test_field_ui_options.py:58]: Hmmmm

File: tests/ui_tests/app/page/common/header.py (hunk 3 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/header.py]: ah, yes, you're right. changed to popup_block

File: tests/ui_tests/test_header_and_footer.py (hunk 6 above)
Comment (madlabman) [reply tests/ui_tests/test_header_and_footer.py:73]: Prepare data using parametrization and or const  -> execute steps in test.
Also it looks like const vars.

Comment (annademina) [reply tests/ui_tests/test_header_and_footer.py:73]: it is this function consts, using parametrization will be a complication and worsed read

**Review by dgusakov — APPROVED** [review]

**Review by skhomuti — APPROVED** [review]

**Review by madlabman — APPROVED** [review]

**Pull request merged**